src/runtime/providers/imagekit.ts

```diff
@@ -171,6 +171,14 @@ export const getImage: ProviderGetImage = (src, { modifiers = {}, baseURL = '/'
   operations = operations.replace('c-pad_extract', 'cm-pad_extract')
   operations = operations.replace('c-extract', 'cm-extract')
 
+  const url = new URL(joinURL(baseURL, src))
+
+  if (url.search) {
+    return {
+      url: joinURL(baseURL, src + (operations ? `&tr=${operations}` : ''))
+    }
+  }
+
   return {
     url: joinURL(baseURL, src + (operations ? `?tr=${operations}` : ''))
```

Comment:
I think here we can use `ufo.withQuery` to append params to the possible existing ones.

Reply:
I have made the requested change.